### PR TITLE
@SafeVarArgs for renderPassThruAttributes

### PR DESCRIPTION
**Pull request by melloware**
@tandraschko after your Sonar cleanup my IDE was warning me to do the above PR.

Can you please review

### CODE REVIEW

File: src/main/java/org/primefaces/util/CalendarUtils.java [hunk 1]
```diff
@@ -245,7 +245,6 @@ public static final String getValue(FacesContext context, UICalendar calendar, O
                     javax.faces.convert.DateTimeConverter nativeConverter = (javax.faces.convert.DateTimeConverter) converter;
                     if (PrimeApplicationContext.getCurrentInstance(context).getEnvironment().isAtLeastJsf23()) {
                         Field field = javax.faces.convert.DateTimeConverter.class.getDeclaredField("type");
-                        field.setAccessible(true);
```
Comment (melloware) [inline src/main/java/org/primefaces/util/CalendarUtils.java]: Sonar recommended removing this accessible set and i verified it was having no effect.

File: src/main/java/org/primefaces/renderkit/CoreRenderer.java [hunk 2]
```diff
@@ -113,8 +110,10 @@ protected void renderPassThruAttributes(FacesContext context, UIComponent compon
         renderDynamicPassThruAttributes(context, component);
     }
 
-    protected void renderPassThruAttributes(FacesContext context, UIComponent component, List<String>... attrs) throws IOException {
+    @SafeVarargs
+    protected final void renderPassThruAttributes(FacesContext context, UIComponent component, List<String>... attrs) throws IOException {
         if (attrs == null) {
+            renderDynamicPassThruAttributes(context, component);
```
Comment (melloware) [inline src/main/java/org/primefaces/renderkit/CoreRenderer.java:116]: If varargs is null the old code still called renderDynamicPassthrough so this puts that back.  Looks at ColorPickerRenderer as an example

**Comment by tandraschko** [comment]
Please dont remove setAccessible. I also wonder why reflection is used over #getType?

**Comment by melloware** [comment]
I use reflection for getType because type is new for JSF 2.3 so to make it backwards compatible reflection must be used to not cause classloader issues.

**Comment by tandraschko** [comment]
Can you recheck this?
https://myfaces.apache.org/core20/myfaces-api/apidocs/javax/faces/convert/DateTimeConverter.html
MF has it since 2.0

**Comment by melloware** [comment]
OK it does look like its always been there but I think the Javadoc is wrong for 2.3.
https://myfaces.apache.org/core23/myfaces-api/apidocs/javax/faces/convert/DateTimeConverter.html#getType--

it says "_Specifies whether the date, time, or both should be parsed/formatted. Values include: date, time, and both. Default based on setting of timeStyle and dateStyle_."

But now the values for 2.3 according to Mojarra:

"_Set the type of value to be formatted or parsed. Valid values are both, date, time localDate, localDateTime, localTime, offsetTime, offsetDateTime, or zonedDateTime. The values starting with "local", "offset" and "zoned" correspond to Java SE 8 Date Time API classes in package java.time with the name derived by upper casing the first letter. For example, java.time.LocalDate for the value "localDate". An invalid value will cause a ConverterException when getAsObject() or getAsString() is called._"

I am going to submit another PR against the original ticket number where I fixed this.

**Pull request merged**